politeiawww/user/localdb/cms.go

```diff
@@ -0,0 +1,237 @@
+	setUser, err := l.UserGetById(nu.ID)
+	if err != nil {
+		return err
+	}
+
+	l.Lock()
+	defer l.Unlock()
+
+	u := user.CMSUser{
+		User:               *setUser,
+		Domain:             nu.Domain,
+		ContractorName:     nu.ContractorName,
+		ContractorType:     nu.ContractorType,
+		ContractorContact:  nu.ContractorContact,
+		ContractorLocation: nu.ContractorLocation,
+		GitHubName:         nu.GitHubName,
+		MatrixName:         nu.MatrixName,
+		ProposalsOwned:     nu.ProposalsOwned,
+		SupervisorUserIDs:  nu.SupervisorUserIDs,
+	}
+	key := []byte(cmsUserPrefix + setUser.Email)
+
+	// Make sure user already exists
+	exists, err := l.userdb.Has(key, nil)
+	if err != nil {
+		return err
+	} else if !exists {
+		return user.ErrUserNotFound
+	}
+
+	payload, err := user.EncodeCMSUser(u)
+	if err != nil {
+		return err
+	}
+
+	return l.userdb.Put(key, payload, nil)
+}
+
+// cmdUpdateCMSUser updates an existing CMSUser record in the database.
+func (l *localdb) cmdUpdateCMSUser(payload string) (string, error) {
+	// Decode payload
+	uu, err := user.DecodeUpdateCMSUser([]byte(payload))
+	if err != nil {
+		return "", err
+	}
+
+	err = l.updateCMSUser(*uu)
+	if err != nil {
+		return "", err
+	}
+
+	// Prepare reply
+	var uur user.UpdateCMSUserReply
+	reply, err := user.EncodeUpdateCMSUserReply(uur)
+	if err != nil {
+		return "", nil
+	}
+
+	return string(reply), nil
+}
+
+// cmdCMSUserByID returns the user information for a given user ID.
+func (l *localdb) cmdCMSUserByID(payload string) (string, error) {
+	// Decode payload
+	p, err := user.DecodeCMSUserByID([]byte(payload))
+	if err != nil {
+		return "", err
+	}
+
+	l.RLock()
+	defer l.RUnlock()
+
+	if l.shutdown {
+		return "", user.ErrShutdown
+	}
+
+	log.Debugf("cmsCMSUserById")
+
+	iter := l.userdb.NewIterator(util.BytesPrefix([]byte(cmsUserPrefix)), nil)
+	for iter.Next() {
+		key := iter.Key()
+		value := iter.Value()
+
+		if !isCMSUserRecord(string(key)) {
+			continue
+		}
+
+		u, err := user.DecodeCMSUser(value)
+		if err != nil {
+			return "", err
+		}
+
+		if u.ID.String() == p.ID {
+			r := user.CMSUserByIDReply{
+				User: u,
+			}
+			reply, err := user.EncodeCMSUserByIDReply(r)
+			if err != nil {
+				return "", err
+			}
+
+			return string(reply), nil
+		}
+	}
+	iter.Release()
+
+	if iter.Error() != nil {
+		return "", iter.Error()
+	}
+
+	return "", user.ErrUserNotFound
+}
+
+// Exec executes a cms plugin command.
+func (l *localdb) cmsPluginExec(cmd, payload string) (string, error) {
+	switch cmd {
+	case user.CmdNewCMSUser:
+		return l.cmdNewCMSUser(payload)
+	case user.CmdUpdateCMSUser:
+		return l.cmdUpdateCMSUser(payload)
+	case user.CmdCMSUserByID:
+		return l.cmdCMSUserByID(payload)
+	default:
+		return "", user.ErrInvalidPluginCmd
+	}
+}
+
+// cmsPluginSetup creates all cms plugin tables and ensures the database
+// is using the correct cms plugin version.
+func (l *localdb) cmsPluginSetup() error {
+
+	// do some kind of leveldb set up here?
```

Comment:
I don't think there's any setup needed for leveldb since its just a simple kv store.